tests/unit/conftest.py

```diff
@@ -1,10 +1,25 @@
 import os
+import random
 
 import docarray
 import pytest
+from tests.constants import HUBBLE_USER_TEST_ID
 
 import hubble
 from finetuner.client import FinetunerV1Client
+from finetuner.constants import (
+    CONFIG,
+    CREATED,
+    CREATED_AT,
+    DESCRIPTION,
+    FAILED,
+    FINISHED,
+    NAME,
+    RUN_NAME,
+    STARTED,
+    STATUS,
+)
+from finetuner.finetuner import Finetuner
```

Comment:
why do we need 2 mock objects? isn't mock_client enough for testing?

Reply:
The first mock object in on the client-level and is used to test how we send requests - checks if url is constructed correctly, if data is passed, etc.

The second mock is on the finetuner-level and is used to imitate what client would've returned to test how creating/deleting/listing things work.

Reply:
The 2 external services that we use in this package is hubble and finetuner-api. So we need 2 mock objects, the first one mocking the hubble SDK and the second one mocking the API. It is difficult to mock the API so you could mock the `FinetunerV1Client` with a mock class that holds the experiments and the runs internally in-memory. I think it is conceptually simpler and more comprehensive this way. For example see how we mock the scheduler in the api